server/api/v1/controllers/quotesApiController.js

```diff
@@ -0,0 +1,279 @@
+"use strict";
+
+const quotesManager = require("../../../../backend/quotes/quotes-manager");
+const moment = require("moment");
+
+exports.getQuotes = async function(req, res) {
+    const quotes = await quotesManager.getAllQuotes();
+
+    const formattedQuotes = quotes.map(q => {
+        return formatQuote(q);
+    });
+
+    res.json(formattedQuotes);
+};
+
+exports.getQuote = async function(req, res) {
+    let { quoteId } = req.params;
+
+    if (quoteId == null) {
+        return res.status(400).send({
+            status: "error",
+            message: "No quoteId provided"
+        });
+    }
+
+    quoteId = parseInt(quoteId);
+    if (isNaN(quoteId)) {
+        return res.status(400).send({
+            status: "error",
+            message: "Invalid quoteId provided"
+        });
+    }
+
+    const quote = await quotesManager.getQuote(quoteId);
+
+    if (quote == null) {
+        return res.status(404).send({
+            status: "error",
+            message: "Quote " + quoteId + " not found"
+        });
+    }
+
+    res.json(formatQuote(quote));
+};
+
+exports.postQuote = async function(req, res) {
+    // Make sure the new quote is valid
+    const validationErrors = validateQuote(req.body);
+
+    if (validationErrors.length > 0) {
+        return res.status(400).send({
+            status: "error",
+            message: validationErrors.join("; ")
+        });
+    }
+
+    try {
+        const quotePost = {
+            text: req.body.text,
+            originator: req.body.originator.replace(/@/g, ""),
+            creator: req.body.creator.replace(/@/g, ""),
+            game: req.body.game,
+            createdAt: moment().toISOString()
+        };
+
+        const newQuoteId = await quotesManager.addQuote(quotePost);
+        const newQuote = formatQuote(await quotesManager.getQuote(newQuoteId));
+
+        return res.status(201).send(newQuote);
+    } catch (e) {
+        return res.status(500).send({
+           status: "error",
+           message: "Error creating quote: " + e
+        });
+    }
+};
+
+exports.putQuote = async function(req, res) {
+    let { quoteId } = req.params;
+    let quotePut = req.body;
```

Comment:
Another `let` that I believe can be a `const` as well

Reply:
Fixed